woocommerce/woocommerce
# CYS: Add Zoomed Out feature

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:

This introduces the following for the **non-AI flow**:

* A device toolbar which allows you to switch between desktop, tablet and mobile
* A zoom out/zoom in button (this switches the device to desktop as thats the only device it works for)
* Disabled the resize handler

Closes https://github.com/woocommerce/woocommerce/issues/50007



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Ensure your CYS experience has been reset.
2. Enter the customize your store flow (non-AI).
3. Check that the device toolbar is present and clicking each device works.
4. Check that clicking zoom in/out resets the device to desktop and also zooms in/out correctly.
5. Check that the resize handler for the frame that was previously present, is no longer enabled.
6. Check that the toolbar to remove/shuffle patterns still works correctly in both zoom in/out mode.
7. Check for no regressions introduced by this PR.
8. Build and upload the plugin to an ecom plan on WP.com and check that **non of these changes are present in the AI flow**.
9. Ensure there are no regressions in the Launch Your Store flow since we updated some shared logic.




### Changelog entry







<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 

Adds zoom out mode to the CYS experience

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>
